### ISSUE TITLE
Crash in pkiSecretBackendConfigUrlsRead()

### ISSUE THREAD

**Issue by danslimmon** [terraform-providers/terraform-provider-vault#598]
### Terraform Version

```
Terraform v0.12.13
+ provider.vault v2.5.0
```

### Affected Resource(s)

- vault_pki_secret_backend_config_urls

### Terraform Configuration Files

```
locals {
  cert_url_map = {
    mirage         = "<snip>"
    oasis          = "<snip>"
    prod-us-east-1 = "<snip>"
  }
}

## mount the PKI secret backend.  The CA used by this backend will have been
## signed by either our internal intermediate environment-specific CA.
resource "vault_mount" "pki" {
  path        = "pki"
  type        = "pki"
  description = "issues certificates"

  default_lease_ttl_seconds = 3 * 24 * 60 * 60 # 3 days
}

## configure the issuing certificates and CRL URLs; the backend doesn't know how
## Vault is reached.
## https://www.vaultproject.io/docs/secrets/pki/index.html#set-url-configuration
resource "vault_pki_secret_backend_config_urls" "config_urls" {
  backend = vault_mount.pki.path

  issuing_certificates = [
    "https://vault.service.consul:8200/v1/${vault_mount.pki.path}/ca",
    "https://vault.service.${terraform.workspace}.consul:8200/v1/${vault_mount.pki.path}/ca",
    "https://${local.cert_url_map[terraform.workspace]}/v1/${vault_mount.pki.path}/ca",
  ]

  crl_distribution_points = [
    "https://vault.service.consul:8200/v1/${vault_mount.pki.path}/crl",
    "https://vault.service.${terraform.workspace}.consul:8200/v1/${vault_mount.pki.path}/crl",
    "https://${local.cert_url_map[terraform.workspace]}/v1/${vault_mount.pki.path}/crl",
  ]
}
```

### Debug Output

https://gist.github.com/danslimmon/a48b99f73aa957b1a7a08555915539b8

### Panic Output

https://gist.github.com/danslimmon/1954420b3ccd9017cb9c12d2e9fe07fe

### Expected Behavior

Terraform should have thrown an error and exited.

### Actual Behavior

Terraform crashed.

### Steps to Reproduce

1. `terraform workspace select mirage`
2. `terraform init`
3. `terraform plan`

### Important Factoids

Nothing I can think of.

### References

Nothing I know of.

**Comment by kalafut** [comment]
@danslimmon Thanks for reporting this. I've reproduced it locally.

**Comment by benpngo** [comment]
@danslimmon 

Experiencing a similar issue as well.

**Comment by kalafut** [comment]
This was recently fixed, though I forgot to reference this issue.

Fixed in #641
